src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -1354,6 +1374,7 @@ public String getUsername() {
      */
     public void setUsernameField(@NotNull String username) {
         this.username = username;
+        this.usernameComponent = null;
```

Comment:
Should probably create the component here, I do not think that it is worth the lazy evaluation